app/javascript/components/rooms/RoomSettings.jsx

```diff
@@ -32,6 +38,25 @@ export default function RoomSettings() {
           <Row>
             <Col className="border-end border-2">
               <h6 className="text-primary">Room Name</h6>
+              <Button
+                variant="primary-light"
+                onClick={() => {
+                  handleGenerateAccessCode();
+                }}
```

Comment:
You can just do `onClick={handleGenerateAccessCode}`

Reply:
It seems to need the arrow function to work.

Reply:
> We should be calling this `viewer_access_code` or `moderator_access_code` since we'll need a way to differentiate them in the future. Just calling it `access_code` means that you'll need to change it anyways in the next PR

I thought we could get away with `access_code` and `mod_access_code` but I agree

Reply:
We've done that before in other files (check `recordingResync`)

Reply:
It doesn't seem to be the same thing.
In recordingResync, the refetch method returned from the useQuery is renamed to handleRecordingResync.
In here, the handleGenerateAccessCode is the mutation itself.
Still investigating this.